lib/ansible/executor/task_executor.py

```diff
@@ -928,14 +928,20 @@ def find_file_in_path(filename):
         # Also need to force a protocol that excludes certain control chars as
         # stdin in this case is a pty and control chars will cause problems.
         # that means only protocol=0 will work.
-        src = cPickle.dumps(self._play_context.serialize(), protocol=0)
-        stdin.write(src)
+        def write_to_socket(obj):
```

Comment:
As a note, we should discourage addition of more nested functions.  Nested functions make testing much hard in the end.

Reply:
I can take it out and make it standalone

Reply:
+1 to this comment.  Move it to a higher level, private function or method